ledger/complete/wal/checkpoint_v6_reader.go

```diff
@@ -83,13 +139,12 @@ func readHeader(filePath string) ([]uint32, uint32, error) {
 // }
 
 func readSubTriesConcurrently(dir string, fileName string, subtrieChecksums []uint32) ([][]*node.Node, error) {
-	// TODO: replace 16 with const
-	if len(subtrieChecksums) != 16 {
-		return nil, fmt.Errorf("expect subtrieChecksums to be %v, but got %v", 16, len(subtrieChecksums))
+	if len(subtrieChecksums) != subtrieCount {
+		return nil, fmt.Errorf("expect subtrieChecksums to be %v, but got %v", subtrieCount, len(subtrieChecksums))
 	}
 
-	resultChs := make([]chan *resultReadSubTrie, 0, 16)
-	for i := 0; i < 16; i++ {
+	resultChs := make([]chan *resultReadSubTrie, 0, subtrieCount)
+	for i := 0; i < subtrieCount; i++ {
 		resultCh := make(chan *resultReadSubTrie)
 		go func(i int) {
 			nodes, err := readCheckpointSubTrie(dir, fileName, i, subtrieChecksums[i])
```

Comment:
question here: if `storeSubTrieConcurrently` can put checksums in arbitrary order(right?), how do we make sure to make the correct mapping btw subtrie filename and the subtrieChecksum? specifically, how are we making sure `subtrieChecksums[i]` does contain the checksum of file `fileName, i`? not sure if i'm missing sth here.

Reply:
specifically, in `storeSubTrieConcurrently`, should one checksum be put into the right slot with `Index`, instead of appending `checksums = append(checksums, result.Checksum)`?

Reply:
@Tonix517 the order of checksums is preserved because:
- `resultChs` is an [array of go channels](https://github.com/onflow/flow-go/blob/30dc86c63f1fb4fdd667646992443ce65066d3b6/ledger/complete/wal/checkpoint_v6_reader.go#L157) in the order of subtries from 0 to 15
- the code [waits and gets results](https://github.com/onflow/flow-go/blob/30dc86c63f1fb4fdd667646992443ce65066d3b6/ledger/complete/wal/checkpoint_v6_reader.go#L162) (including checksum) from each go channel in order
- results are [appended](https://github.com/onflow/flow-go/blob/30dc86c63f1fb4fdd667646992443ce65066d3b6/ledger/complete/wal/checkpoint_v6_reader.go#L167) to `nodesGroups` in order.

Reply:
>if storeSubTrieConcurrently can put checksums in arbitrary order(right?)

No , because we append the result to `resultChs` the same order as the subtrie group.  see L157

And even if some resultChs is completed earlier, the result won't be consumed. The results for each group is consumed the same order as subtrie group (see L162)